Client/documentation_portal/src/components/documents/ExcelViewer.tsx

```
@@ -100,7 +156,18 @@ export function ExcelViewer({ fileUrl }: ExcelViewerProps) {
   const headers = filteredRows[0] || [];
-  const rows = filteredRows.slice(1);
+  const allRows = filteredRows.slice(1);
+  
+  // Pagination - only render visible rows to prevent memory issues
+  const totalPages = Math.ceil(allRows.length / rowsPerPage);
+  const startIndex = (currentPage - 1) * rowsPerPage;
+  const endIndex = startIndex + rowsPerPage;
+  const rows = allRows.slice(startIndex, endIndex);
+  
+  // Reset to page 1 when sheet changes or search changes
+  useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)